experimental/generation/generator/templates/professional/library-ChooseIntent.dialog.lg

```diff
@@ -0,0 +1,59 @@
+# template
+- ```
+{
+    "$schema": "${appSchema}",
+    "$kind": "Microsoft.OnChooseIntent",
+    "actions": [
+        {
+            "$kind": "Microsoft.SetProperty",
+            "property": "dialog.recognized",
+            "value": "=turn.recognized"
+        },
```

Comment:
Why is this needed?

Reply:
When the utterance is recognized by luis and qnamaker with the fair score, we need the user to choose intent.

Reply:
I see this is persisting so you can use it for the final result.  
To switch this to an ask.  (Not needed now.)
1) Have this trigger set dialog.chooseRecognizer=turn.recognized
2) Have a separate trigger that is triggered by the presence of dialog.chooseRecognizer and has a 0 priority.  It should do the ask.
3) Setup the LUIS models to detect the different sources and a way to say none of them.
4) When one of the selections happens, have a trigger which selects the appropriate one and clear dialog.chooseIntent

---
In reply to: [435744407](https://github.com/microsoft/BotBuilder-Samples/pull/2474#discussion_r435744407) [](ancestors = 435744407)